storage-provider/server/src/p2p/register.rs

```
@@ -59,83 +61,98 @@ impl RegisterConfig {
+                } else {
+                    tracing::info!("Registration with {rendezvous_point} successful");
                 }
             }
-            SwarmEvent::Behaviour(RegisterBehaviourEvent::Rendezvous(
-                rendezvous::client::Event::Registered {
-                    namespace,
-                    ttl,
-                    rendezvous_node,
-                },
-            )) => {
-                tracing::info!(
-                    "Registered for namespace '{}' at rendezvous point {} for the next {} seconds",
-                    namespace,
-                    rendezvous_node,
-                    ttl
-                );
-                return Ok(());
-            }
-            SwarmEvent::Behaviour(RegisterBehaviourEvent::Rendezvous(
-                rendezvous::client::Event::RegisterFailed {
-                    rendezvous_node,
-                    namespace,
-                    error,
-                },
+            // Check incoming event.
+            event = swarm.select_next_some() => check_swarm_event(event)
+        }
+    }
+}
+
+fn check_swarm_event(event: SwarmEvent<RegisterBehaviourEvent>) {
+    match event {
+        SwarmEvent::Behaviour(RegisterBehaviourEvent::Rendezvous(
+            rendezvous::client::Event::Registered {
+                namespace,
+                ttl,
+                rendezvous_node,
+            },
+        )) => {
+            tracing::info!(
+                "Registered for namespace '{}' at rendezvous point {} for the next {} seconds",
+                namespace,
+                rendezvous_node,
+                ttl
+            );
+        }
+        SwarmEvent::Behaviour(RegisterBehaviourEvent::Rendezvous(
+            rendezvous::client::Event::RegisterFailed {
+                rendezvous_node,
+                namespace,
+                error,
+            },
+        )) => {
+            tracing::error!(%rendezvous_node, %namespace,
```

Should we return an error here?